src/lib/comms/sol-http-server-impl-microhttpd.c

```diff
@@ -270,6 +275,100 @@ build_mhd_response(const struct sol_http_response *response, time_t last_modifie
     return NULL;
 }
 
+static struct MHD_Response *
+build_mhd_response(const struct sol_http_response *response, time_t last_modified)
+{
+    struct MHD_Response *r;
+
+    r = MHD_create_response_from_buffer(response->content.used, response->content.data,
+        MHD_RESPMEM_MUST_COPY);
+    if (!r)
+        return NULL;
+
+    return build_mhd_response_params(r, response, last_modified);
+}
+
+static void
+response_sse_del_cb(void *data)
+{
+    struct sol_http_response_sse *sse = data;
+
+    if (sse->cb)
+        sse->cb((void *)sse->cb_data, sse);
+
+    sol_buffer_fini(&sse->buffer);
+    free(sse);
+}
+
+static ssize_t
+response_sse_cb(void *data, uint64_t pos, char *buf, size_t size)
+{
+    int r;
+    ssize_t len;
+    struct sol_http_response_sse *sse = data;
+
+    if (sse->delete_me)
+        return MHD_CONTENT_READER_END_OF_STREAM;
+
+    if (!sse->buffer.used) {
+        MHD_suspend_connection(sse->connection);
+        sse->suspended = true;
+        return 0;
+    }
+
+    len = sse->buffer.used > size ? size : sse->buffer.used;
+    memcpy(buf, sse->buffer.data, len);
```

Comment:
as we have talked, this buffer is given by microhttpd and there is no way (now) to we pass our own buffer, so the copy is necessary.